h2o-core/src/test/java/water/UnknownHeartbeatTest.java

```diff
@@ -17,12 +16,43 @@ public void testIgnoreUnknownHeartBeat() {
     hb._client = true;
     hb._jar_md5 = H2O.SELF._heartbeat._jar_md5;
 
-    AutoBuffer ab = new AutoBuffer(H2O.SELF, H2O.MAX_PRIORITY);
+    AutoBuffer ab = new AutoBuffer(H2O.SELF, UDP.udp.heartbeat._prior);
     ab.putUdp(UDP.udp.heartbeat, 65400); // put different port number to simulate heartbeat from fake node
     hb.write(ab);
     ab.close();
 
     // Verify that we don't have a new client
     assertEquals(0, H2O.getClients().size());
   }
+
+  @Test
+  public void testIgnoreUnknownShutdownTask(){
+    AutoBuffer ab = new AutoBuffer(H2O.SELF, UDP.udp.rebooted._prior);
```

Comment:
For these kind of tests it would be really helpful to have some kind of hooks that would notify us that the heartbeat was discarded.

Reply:
That would require a bigger change of the core internals, however yes, it would be useful.

Though for now, I think it's fine to check that the post-condition triggered by the wrong heartbeat didn't not occur. For example by checking number of connected clients.